src/plugins/usage_collection/server/routes/report_metrics.ts

```diff
@@ -48,15 +53,97 @@ export function registerUiMetricRoute(
     async (context, req, res) => {
       const { report } = req.body;
       try {
-        const internalRepository = getSavedObjects();
-        if (!internalRepository) {
-          throw Error(`The saved objects client hasn't been initialised yet`);
+        const currTime = Date.now();
+
+        // Add the current report to batchReport
+        batchReport.report = combineReports(report, batchReport.report);
+        // If the time duration since the batchReport startTime is greater than batchInterval then write it to the savedObject
+        if (currTime - batchReport.startTimestamp >= batchingIntervalInMs) {
+          const prevReport = batchReport;
+
+          batchReport = {
+            report: {},
+            startTimestamp: currTime,
+          }; // reseting the batchReport and updating the startTimestamp to current TimeStamp
+
+          if (prevReport) {
+            // Write the previously batched Report to the saved object
+            const internalRepository = getSavedObjects();
+            if (!internalRepository) {
+              throw Error(`The saved objects client hasn't been initialised yet`);
+            }
+            await storeReport(internalRepository, prevReport.report);
+          }
         }
-        await storeReport(internalRepository, report);
+
         return res.ok({ body: { status: 'ok' } });
       } catch (error) {
         return res.ok({ body: { status: 'fail' } });
       }
     }
   );
 }
+
+function combineReports(report1: ReportSchemaType, report2: ReportSchemaType) {
+  // Combines report2 onto the report1 and returns the updated report1
+
+  // Combining User Agents
+  const combinedUserAgent = { ...report1.userAgent };
+
+  if (report2.userAgent !== undefined) {
+    for (const key in Object.keys(report2.userAgent)) {
+      if (report2.userAgent[key] === undefined) {
+        continue;
+      }
+
+      // Only adding the useragents not already present in report1
+      if (!report1.userAgent || report1.userAgent[key] === undefined) {
+        combinedUserAgent[key] = report2.userAgent[key];
+      }
+    }
+  }
```

Comment:
Why not just do
```suggestion
  const combinedUserAgent = { ...report2.userAgent , ...report1.userAgent };
```

Reply:
Nice Suggestion. Updated it to use the above style.